### ISSUE TITLE
--desktop requires xhost to be used on Solus

### ISSUE THREAD

**Issue by UniversalSuperBox** [bhdouglass/clickable#26]
In order to get an app to run with `--desktop` on my computer running Solus, I had to install xhost then do the following:

```
xhost local:root
```

I was then able to run the app normally.

Before doing this, I was getting the following error: https://paste.ubuntu.com/26388791/

Edit: The command needs to be repeated on every reboot. Subsequent runs will still exit with status 0, so running it every run of Clickable could work.

**Comment by bhdouglass** [comment]
Any idea if this is just a Solus issue or is it an issue with other distros as well (like ubuntu?).

**Comment by delijati** [comment]
Not sure, i have also `xhost` on my ubuntu 17.10 installed but i'm not calling or setting it in any way.

**Comment by bhdouglass** [comment]
Thanks @delijati must not be a problem on Ubuntu. I'll see what I can do to make this work well for Solus.

**Comment by userj** [comment]
I am getting this error on Ubuntu though:
```
Skipping install, running in desktop mode
QXcbConnection: Could not connect to display :1
```

**Comment by bhdouglass** [comment]
@userj what xdisplay are you using? clickable will use whatever is in your `DISPLAY` env var.

**Comment by userj** [comment]
@bhdouglass 
I'm not totally sure if I'm doing this correctly, but on Ubuntu 17.10 (xorg) I have these results:

```
echo $DISPLAY
:1
```
I tried running 
`DISPLAY=:0 clickable --desktop` and I get the same "could not connect" error.

Also 
`xhost local:root` did not seem to fix it for me.

**Comment by bhdouglass** [comment]
@userj you're not using wayland or anything are you?

**Comment by userj** [comment]
@bhdouglass I am using Ubuntu Xorg, as selected on the login menu. HOWEVER, I noticed that the "Ubuntu" (Wayland) option is not even there. Instead I have a "Unity" option for some reason. I think I tried to install that once upon a time.

I have tested the --desktop flag on another computer and it worked without an issue, so something is funky on the computer mentioned here.

**Comment by bhdouglass** [comment]
That's weird, I'm not even sure how you would go about trying to figure out what went wrong! But it seems like it's not a problem with clickable so that's good.

**Comment by mrliau** [comment]
Btw, I resolved this issue by uninstalling and reinstalling clickable.

**Comment by userj** [comment]
I was able to get it working by uninstalling and reinstalling clickable.

**Comment by bhdouglass** [comment]
awesome, glad you were able to resolve that!

**Comment by userj** [comment]
@bhdouglass Yes, but then it also stopped working again after reboot, which is why it sounds similar to @UniversalSuperBox's issue. But then the `xhost local:root` didn't work.

**Comment by bhdouglass** [comment]
@userj that's very strange

**Comment by bhdouglass** [comment]
I've added a call to xhost to fix this issue (if xhost is insatlled) in v4.1.0 (in the ppa soon!)
